relayer/relays/beacon/header/syncer/syncer.go

```diff
@@ -117,16 +117,18 @@ func (s *Syncer) GetSyncCommitteePeriodUpdate(from uint64) (scale.Update, error)
 		return scale.Update{}, fmt.Errorf("parse signature slot as int: %w", err)
 	}
 
-	blockRootsProof, err := s.GetBlockRoots(uint64(finalizedHeader.Slot))
-	if err != nil {
-		return scale.Update{}, fmt.Errorf("fetch block roots: %w", err)
-	}
-
 	finalizedHeaderBlockRoot, err := finalizedHeader.ToSSZ().HashTreeRoot()
 	if err != nil {
 		return scale.Update{}, fmt.Errorf("beacon header hash tree root: %w", err)
 	}
 
+	blockRootsProof, err := s.GetBlockRoots(uint64(finalizedHeader.Slot))
+	// It's possible that beaconState is not available when GetSyncCommitteePeriodUpdate, in this case
+	// there is no blockRootsProof and just ignore it in beacon client
+	if err != nil && err != ErrBeaconStateAvailableYet {
+		return scale.Update{}, fmt.Errorf("fetch block roots: %w", err)
+	}
```

Comment:
Allow SyncCommiteeUpdate without blockRootsProof

Reply:
I'm not sure this change is a good one. Does it even make sense to send the finalized header update without an ancestry proof? That's the main reason we send beacon header updates in the first place.

Reply:
It's an edge case that I run `generateBeaconTestFixture` for the `SyncCommitteePeriodUpdate` from sepolia and beaconState is not available at that moment. Code in https://github.com/Snowfork/snowbridge/blob/faacc1e1cb43f5e04b9dae059453d3da717c8cc2/relayer/cmd/generate_beacon_data.go#L189-L193

But I think it's possible to switch to another time to generate the test fixture and as you mentioned it could lead to some unexpected behavior to relax the check so just revert it.